frontend/public/components/modals/delete-pvc-modal.tsx

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react';
+import { Stack, StackItem } from '@patternfly/react-core';
+import { AsyncComponent, HandlePromiseProps, withHandlePromise } from '../utils';
+import { getName, YellowExclamationTriangleIcon } from '@console/shared';
+import {
+  createModalLauncher,
+  ModalTitle,
+  ModalBody,
+  ModalSubmitFooter,
+  ModalComponentProps,
+} from '../factory';
+import { k8sKill, K8sResourceCommon } from '@console/internal/module/k8s';
+import { PersistentVolumeClaimModel } from '../../models';
+import { isPVCDelete, PVCDelete, useExtensions } from '@console/plugin-sdk';
+
+const DeletePVCModal = withHandlePromise((props: DeletePVCModalProps) => {
+  const { pvc, inProgress, errorMessage, handlePromise, close, cancel } = props;
+  const pvcDeleteExtensions = useExtensions<PVCDelete>(isPVCDelete);
+  const pvcName = getName(pvc);
+
+  const submit = (e) => {
+    e.preventDefault();
+
+    const promise = k8sKill(PersistentVolumeClaimModel, pvc).then(() =>
+      Promise.all(
+        pvcDeleteExtensions.map(
+          ({ properties: { predicate, onPVCKill } }) => predicate(pvc) && onPVCKill(pvc),
+        ),
+      ),
+    );
+    return handlePromise(promise, close);
+  };
+
+  const alertComponents = pvcDeleteExtensions.map(
+    ({ properties: { predicate, alertLoader } }, i) =>
+      predicate(pvc) && <AsyncComponent key={`ext-${i}`} loader={alertLoader} pvc={pvc} />,
+  );
+
+  return (
+    <form onSubmit={submit} className="modal-content">
+      <ModalTitle>
+        <YellowExclamationTriangleIcon className="co-icon-space-r" /> Delete Persistent Volume Claim
+      </ModalTitle>
+      <ModalBody>
+        <Stack hasGutter>
+          {alertComponents.map((alert, i) => (
```

Comment:
```suggestion
          {alertComponents}
```

Reply:
I want to render a margin between every alert `<StackItem>`, so I need this map.

Reply:
See my comment https://github.com/openshift/console/pull/6407#discussion_r480074555